crates/router/src/core/utils.rs

```diff
@@ -2018,3 +2022,80 @@ pub(crate) fn validate_profile_id_from_auth_layer<T: GetProfileId + std::fmt::De
         (None, None) | (None, Some(_)) => Ok(()),
     }
 }
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+pub async fn construct_vault_router_data<F>(
```

Comment:
We can use v2 router instead of using v1 router, if execute required v1 router_data already there is a function to do the conversion, so that core flow there will be no changes, only a generic transformer is required to do the conversion between routerdatas